src/braket/default_simulator/openqasm/parser/braket_pragmas.py

```diff
@@ -209,11 +223,10 @@ def visitKraus(self, ctx: BraketPragmasParser.KrausContext):
         matrices = [self.visit(m) for m in ctx.matrices().children[::2]]
         return Kraus(target, matrices)
 
-    def visitProbabilities(self, ctx: BraketPragmasParser.ProbabilitiesContext):
-        return [float(prob.symbol.text) for prob in ctx.children[::2]]
-
 
-def parse_braket_pragma(pragma_body: str, qubit_table: "QubitTable"):
+def parse_braket_pragma(
+    pragma_body: str, qubit_table: "QubitTable", pragma_node_visitor=BraketPragmaNodeVisitor
```

Comment:
Can we pass in the node visitor object here instead of the class? The way we're currently passing this in couples the node visitor to the implementation and the information available in this method.

Reply:
parse_braket_pragma function initializes a node visitor using the second param quibit_table to initialize an object and then call function called visit using derived data from pragma_body.

Background: 
This part of the code can be made generic by moving this to program context. But for the sake of backward compatibility I had to create an abstract class. So a generic class can be created when we are using it in different places like braket-sdk or qsim.

Reply:
Thanks for the background. What I'm trying to call out is that in case in the future BraketPragmaNodeVisitor needs more inputs to initialize, you'd need to pass them all in as args here instead of initializing the BraketPragmaNodeVisitor and passing that in. If there are different strategies for initializing BraketPragmaNodeVisitor, those would need to be duplicated within this method.